src/aiida/orm/nodes/caching.py

```diff
@@ -44,7 +45,7 @@ def _get_hash(self, ignore_errors: bool = True, **kwargs: t.Any) -> str | None:
         :param ignore_errors: return ``None`` on ``aiida.common.exceptions.HashingError`` (logging the exception)
         """
         try:
-            return make_hash(self._get_objects_to_hash(), **kwargs)
+            return make_hash(self.get_objects_to_hash(), **kwargs)
```

Comment:
@danielhollas thanks for dig this out. If I understand it correctly, initial scope was not changing the hashing. So for this line, I think it needs convert the dict into the original format (which is the list of values of the dict) that passing to `make_hash`.

Reply:
This was not a mistake. We intentionally changed this now, because there were anyway a couple of other fixes that changed hashes, and these were all released together in 2.6.0. See also the [entry in the CHANGELOG](https://github.com/aiidateam/aiida-core/blob/v2.6.0/CHANGELOG.md#improvements-and-changes-to-caching) that explains this and tells users that nodes should be rehashed. So I wouldn't revert this.

Reply:
Okay. But what you expect to be tested? Something that the hashing should changed after the change?

Reply:
I think we should have tests that have hardcoded hash values, so that we're always aware that the hashes have changed. There was for example a cases were there was a new attribute  added on the calcjob class and they were by mistake not included in the list of ignored attributes (I'll find the PR later).

Reply:
Also, @sphuber is right, we can't revert now even if this wasn't intentional, since then we'd have another hash invalidation.